cmd/bridge/downloads/server.go

```diff
@@ -0,0 +1,258 @@
+		},
+		{
+			arch:            "arm64",
+			operatingSystem: "linux",
+			path:            "/usr/share/openshift/linux_arm64/oc",
+		},
+		{
+			arch:            "arm64",
+			operatingSystem: "mac",
+			path:            "/usr/share/openshift/mac_arm64/oc",
+		},
+		{
+			arch:            "ppc64le",
+			operatingSystem: "linux",
+			path:            "/usr/share/openshift/linux_ppc64le/oc",
+		},
+		{
+			arch:            "s390x",
+			operatingSystem: "linux",
+			path:            "/usr/share/openshift/linux_s390x/oc",
+		},
+	}
+
+	for _, spec := range specs {
+		basename := filepath.Base(spec.path)
+		targetPath := filepath.Join(spec.arch, spec.operatingSystem, basename)
+		dirPath := filepath.Join(spec.arch, spec.operatingSystem)
+		os.Mkdir(dirPath, 0755)
+		os.Symlink(spec.path, targetPath)
+		//extension := filepath.Ext(spec.path)
+		//baseRoot := basename[:len(basename)-len(extension)]
+		archivePathRoot := filepath.Join(spec.arch, spec.operatingSystem, basename)
+
+		//fmt.Printf("path: %s\n", spec.path)
+		//fmt.Printf("target: %s\n", targetPath)
+		//fmt.Printf("dir: %s\n", dirPath)
+		//fmt.Printf("Base name without extension (baseRoot): %s\n", baseRoot)
+    	//fmt.Printf("Extension: %s\n\n", extension)
+		//fmt.Printf("Archive path root: %s\n", archivePathRoot)
+		createArchive(archivePathRoot + ".tar", targetPath)
+		content = append(content, fmt.Sprintf("<a href=\"%s\">oc (%s %s)</a> (<a href=\"%s.tar\">tar</a> <a href=\"%s.zip\">zip</a>)",
+        targetPath, spec.arch, spec.operatingSystem, archivePathRoot, archivePathRoot))
+	}
+	fmt.Println(content)
+	
+	// Walk the temporary directory and create index files for each directory
+	err = filepath.Walk(tempDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			// Get the relative path from tempDir to the current directory
+			joinedChilds := filepath.Join(path, "child")
+
+			relPath, err := filepath.Rel(tempDir, joinedChilds)
+			if err != nil {
+				return err
+			}
+
+			// Replace path separators with '/'
+			rootLink := strings.Replace(relPath, string(filepath.Separator), "/", -1)
+
+			// Write index for the current directory
+			writeIndex(
+				filepath.Join(path, "index.html"),
+				fmt.Sprintf("<p>Directory listings are disabled. See <a href=\"%s\">here</a> for available content.</p>", rootLink),
+			)
+		}
+
+		return nil
+	})
+	if err != nil {	
+		fmt.Println("Error walking the path : ", err)
+	}
+
+	// construct the list of links
+	message := "<ul>\n"
+	for _, entry := range content {
+		message += (fmt.Sprintf("  <li>%s</li>\n", entry))
+	}
+	message += "</ul>\n"
+
+	// Write the index file
+	err = writeIndex(filepath.Join(tempDir, "index.html"), message)
+	if err != nil {
+		fmt.Println("Error writing index file: ", err)
+	}
+	
+	// Serve files from the temporary directory
+    http.Handle("/", http.FileServer(http.Dir(tempDir)))
+
+    // Define the address to listen on
+    addr := ":8080"
```

Comment:
dynamic-demo-plugin is already using this port. Lets use some other port in order to prevent any isues/confusion.

Reply:
Good point. I changed it to 8081 for now. I'm not sure if I should go for something more distinct, to avoid some potentional port conflict issues?